sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -265,6 +265,9 @@ class SessionCatalog(
     if (dbName == DEFAULT_DATABASE) {
       throw QueryCompilationErrors.cannotDropDefaultDatabaseError
     }
+    if (!ignoreIfNotExists) {
+      requireDbExists(dbName)
```

Comment:
My only concern is that now `externalCatalog` cannot handle `ignoreIfNotExists` properly. Previously the handling was delegated.

Reply:
Not sure I get your ponit. You mean this `SessionCatalog` can not handle `ignoreIfNotExists` correctly ?

Reply:
I mean, if the DB already exists, `externalCatalog` cannot handle the case anymore because it will throw an exception in the Spark. Previously it was delegated to `externalCatalog` right?

Reply:
`externalCatalog` can be not only Hive but can be other implementations, and maybe there could be some special handling for these cases (e.g., cleaning up resources)

Reply:
thanks for the explaination, I see your concern. Seems we also didn't handle this in `drop table` or `drop function`.. 
```
requireDbExists(db)
if (tableExists(TableIdentifier(table, Option(db)))) {
  externalCatalog.dropTable(db, table, ignoreIfNotExists = true, purge = purge)
} else if (!ignoreIfNotExists) {
  throw new NoSuchTableException(db = db, table = table)
}
```

So I'm assuming it's OK in `drop database` ?

Reply:
Hm, okie.

Reply:
hm .. okie

Reply:
As I see it, we'd better remove `requireDbExists` check from the table and function APIs in the `SessionCatalog`?